Essentials/src/tpr.yml

```diff
@@ -0,0 +1,16 @@
+# Configuration for the random teleport command.
+# Some settings may be defaulted, and can be changed via the /settpr command in-game.
```

Comment:
Shouldn't this file include all the available config options (ie pre-cache, max-range)?

Reply:
Arguably yes, although once something gets saved into the file, it is effectively permanent. This can be problematic, so unless the server owner explicitly sets some of these settings. Here's an explanation for the thought process behind putting all of the current settings where they are:

`center`: We cannot have a default in the config, because we cannot assume what the world name is. So instead, when `getCenter` is called, we try reading the config first, but if that isn't available (or the world is no longer valid), we write the config with the center of the current main world, which is an appropriate default. This can further be changed via explicitly running the `/settpr` command or entering manual config.

`min-range`: In the config by default, and also automatically zero if none specified. Zero is a fine default, and the server owner can increase it if they do not want players teleporting into a spawn protected zone, etc.

`max-range`: Not explicitly in the config, because we want the default to be the max range of whatever world the command is being used in. This allows a true random location in the world, when server owners do not specify something lower. We don't want to hard code 3E7 (the default vanilla world border size) into here, because the server owner may have already changed it. It can be lowered if servers don't want a true random location, but rather something closer to spawn (like 1k, 10k, etc).

`excluded-biomes`: Which biomes should be avoided when selecting a random location. Safe teleport doesn't handle this, and will happily randomly teleport players into oceans for example (and fairly often actually) if not otherwise checked. This is set in the config by default since there is something sensible to set it to, and the server owner can adjust it if they want to add/remove anything.

`find-attempts`: Used internally for determining how many times to try finding random locations. This is something that generally should not be changed (higher = potentially more lag, lower = potentially not finding locations). Server owners can adjust it to tweak performance vs optimal behavior, but it's better not written in the config by default in case Essentials wants to adjust the default value in the future. #EasterEgg 

`cache-threshold`: Also used internally to decide when the cache is getting too empty, and needs to be refilled. Not really much reason to need to change this either, in most cases. #EasterEgg

`pre-cache`: Having this on by default makes the command run much faster (particularly, on the first execution, where there may otherwise be no cache). Having a small number of locations cached isn't particularly heavy, so there is not much reason for most server owners to touch this. It still exists as an option to disable though. #EasterEgg